tests run only in a build folder, that is one level deeper in the directory tree than the source (like the provided build folder). Building anywhere else fails:

```
gcc -DPACKAGE_NAME=\"ROFL-DATAPATH\" -DPACKAGE_TARNAME=\"rofl-datapath\" -DPACKAGE_VERSION=\"v0.6.4\" -DPACKAGE_STRING=\"ROFL-DATAPATH\ v0.6.4\" -DPACKAGE_BUGREPORT=\"rofl-devel@roflibs.org\" -DPACKAGE_URL=\"http://www.roflibs.org\" -DPACKAGE=\"rofl-datapath\" -DVERSION=\"v0.6.4\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -D__EXTENSIONS__=1 -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D_POSIX_PTHREAD_SEMANTICS=1 -D_TANDEM_SOURCE=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -D__STDC_FORMAT_MACROS=/\*\*/ -DDEBUG=/\*\*/ -DHAVE_MA_TRIE=1 -DHAVE_MA_LOOP=1 -DHAVE_MA_L2HASH=1 -I. -I/home/tobi/workspace/rofl-datapath/src/ -I/home/tobi/workspace/rofl-datapath/src/ -I../src/ -I/home/tobi/workspace/rofl-datapath/test/rofl/datapath/pipeline/../src/ -I/home/tobi/workspace/rofl-datapath/test/rofl/datapath/pipeline/../src/ -I/home/tobi/workspace/rofl-datapath/test/rofl/datapath/pipeline/../src/ -I/home/tobi/workspace/rofl-datapath/test/rofl/datapath/pipeline/../src/ -DROFL_TEST=1   -std=gnu89 -Werror -Wall  -fno-omit-frame-pointer -g -g -O0 -D__COMPILING_ROFL_PIPELINE__ -c -o openflow/openflow1x/pipeline/memory.o openflow/openflow1x/pipeline/memory.c
tern_alike_test.c:6:77: fatal error: ../../../../../src/rofl/datapath/pipeline/common/ternary_fields.h: No such file or directory
 #include "../../../../../src/rofl/datapath/pipeline/common/ternary_fields.h"
                                                                             ^
```

--- a/src/rofl/datapath/pipeline/openflow/openflow1x/pipeline/matching_algorithms/trie/of1x_trie_ma.c
+++ b/src/rofl/datapath/pipeline/openflow/openflow1x/pipeline/matching_algorithms/trie/of1x_trie_ma.c
@@ -618,6 +618,13 @@ rofl_of1x_fm_result_t __of1x_add_leafs_trie(of1x_trie_t* trie,
 		}
 
 ADD_LEAFS_INNER:
+		if(!l->inner){
+			//This is the point of insertion
+			res = __of1x_insert_intermediate_leaf_trie(trie, l,
+									m_it,
+									entry);
+			goto ADD_LEAFS_END;
+		}
 		l = l->inner;
 		continue;
 

--- a/test/rofl/datapath/pipeline/openflow/openflow1x/pipeline/ma/trie/trie.c
+++ b/test/rofl/datapath/pipeline/openflow/openflow1x/pipeline/ma/trie/trie.c
@@ -978,16 +978,6 @@ void test_regression1(){
 	entry = of1x_init_flow_entry(false);
 	CU_ASSERT(entry != NULL);
 
-#if 0
-	CU_ASSERT(of1x_add_match_to_entry(entry,of1x_init_eth_src_match(0xAABBCCDDEEFF, 0xFFF0FFFFFFFF)) == ROFL_SUCCESS);
-	CU_ASSERT(of1x_add_match_to_entry(entry,of1x_init_ip4_dst_match(0xC0A80001, 0xFFFFFFFF)) == ROFL_SUCCESS);
-	CU_ASSERT(of1x_add_match_to_entry(entry,of1x_init_ip4_src_match(0xC0A80002, 0xFFFFFFFF)) == ROFL_SUCCESS);
-	entry->priority=99; //Same exact priority => replace
-	CU_ASSERT(of1x_add_flow_entry_table(&sw->pipeline, 0, &entry, false,false) == ROFL_OF1X_FM_SUCCESS);
-
-	of1x_full_dump_switch(sw, false);
-	CU_ASSERT(table->num_of_entries == 2);
-#endif
 	CU_ASSERT(of1x_add_match_to_entry(entry,of1x_init_ip4_dst_match(0xC0A80001, 0xFFFFFFFF)) == ROFL_SUCCESS);
 	entry->priority=31; //Same exact priority => replace
 	CU_ASSERT(of1x_add_flow_entry_table(&sw->pipeline, 0, &entry, false,false) == ROFL_OF1X_FM_SUCCESS);

--- a/test/rofl/datapath/pipeline/openflow/openflow1x/pipeline/ma/trie/trie.c
+++ b/test/rofl/datapath/pipeline/openflow/openflow1x/pipeline/ma/trie/trie.c
@@ -958,6 +958,57 @@ void test_regressions(){
 
 }
 
+void test_regression1(){
+
+	of1x_flow_entry_t* entry;
+
+	//First entry
+	clean_all();
+	entry = of1x_init_flow_entry(false);
+	CU_ASSERT(entry != NULL);
+
+	CU_ASSERT(of1x_add_match_to_entry(entry,of1x_init_ip4_dst_match(0xC0A80000, 0xFFFFFF00)) == ROFL_SUCCESS);
+	entry->priority=32;
+	CU_ASSERT(of1x_add_flow_entry_table(&sw->pipeline, 0, &entry, false,false) == ROFL_OF1X_FM_SUCCESS);
+
+	of1x_full_dump_switch(sw, false);
+	CU_ASSERT(table->num_of_entries == 1);
+
+	//Add one with the same + 1 match
+	entry = of1x_init_flow_entry(false);
+	CU_ASSERT(entry != NULL);
+
+#if 0
+	CU_ASSERT(of1x_add_match_to_entry(entry,of1x_init_eth_src_match(0xAABBCCDDEEFF, 0xFFF0FFFFFFFF)) == ROFL_SUCCESS);
+	CU_ASSERT(of1x_add_match_to_entry(entry,of1x_init_ip4_dst_match(0xC0A80001, 0xFFFFFFFF)) == ROFL_SUCCESS);
+	CU_ASSERT(of1x_add_match_to_entry(entry,of1x_init_ip4_src_match(0xC0A80002, 0xFFFFFFFF)) == ROFL_SUCCESS);
+	entry->priority=99; //Same exact priority => replace
+	CU_ASSERT(of1x_add_flow_entry_table(&sw->pipeline, 0, &entry, false,false) == ROFL_OF1X_FM_SUCCESS);
+
+	of1x_full_dump_switch(sw, false);
+	CU_ASSERT(table->num_of_entries == 2);
+#endif
+	CU_ASSERT(of1x_add_match_to_entry(entry,of1x_init_ip4_dst_match(0xC0A80001, 0xFFFFFFFF)) == ROFL_SUCCESS);
+	entry->priority=31; //Same exact priority => replace
+	CU_ASSERT(of1x_add_flow_entry_table(&sw->pipeline, 0, &entry, false,false) == ROFL_OF1X_FM_SUCCESS);
+
+	of1x_full_dump_switch(sw, false);
+	CU_ASSERT(table->num_of_entries == 2);
+
+	//Remove intermediate, intermediate should be pruned
+	entry = of1x_init_flow_entry(false);
+	CU_ASSERT(of1x_add_match_to_entry(entry,of1x_init_ip4_dst_match(0xC0A80000, 0xFFFFFF00)) == ROFL_SUCCESS);
+	CU_ASSERT(of1x_remove_flow_entry_table(&sw->pipeline, 0, entry, false, OF1X_PORT_ANY, OF1X_GROUP_ANY) == ROFL_OF1X_FM_SUCCESS);
+	of1x_full_dump_switch(sw, false);
+	CU_ASSERT(table->num_of_entries == 1);
+
+	entry = of1x_init_flow_entry(false);
+	CU_ASSERT(entry != NULL);
+	CU_ASSERT(of1x_remove_flow_entry_table(&sw->pipeline, 0, entry, false, OF1X_PORT_ANY, OF1X_GROUP_ANY) == ROFL_OF1X_FM_SUCCESS);
+
+	of1x_full_dump_switch(sw, false);
+	CU_ASSERT(table->num_of_entries == 0);
+}
 
 #define NUM_ENTRIES 260
 void test_many_entries(){

--- a/test/rofl/datapath/pipeline/openflow/openflow1x/pipeline/ma/trie/trie.h
+++ b/test/rofl/datapath/pipeline/openflow/openflow1x/pipeline/ma/trie/trie.h
@@ -24,5 +24,6 @@ void test_install_flowmods(void);
 void test_remove_flowmods(void);
 void test_many_entries(void);
 void test_regressions(void);
+void test_regression1(void);
 
 #endif

--- a/test/rofl/datapath/pipeline/openflow/openflow1x/pipeline/ma/trie/unit_test.c
+++ b/test/rofl/datapath/pipeline/openflow/openflow1x/pipeline/ma/trie/unit_test.c
@@ -28,7 +28,8 @@ int main(int args, char** argv){
 	(NULL == CU_add_test(pSuite, "Trie: test install regular flowmods without and without masks", test_install_flowmods)) ||
 	(NULL == CU_add_test(pSuite, "Trie: test removal regular flowmods without and without masks", test_remove_flowmods)) ||
 	(NULL == CU_add_test(pSuite, "Trie: test many entries", test_many_entries)) ||
-	(NULL == CU_add_test(pSuite, "Trie: test regressions", test_regressions)) //||
+	(NULL == CU_add_test(pSuite, "Trie: test regressions", test_regressions)) ||
+	(NULL == CU_add_test(pSuite, "Trie: test regressions 1", test_regression1)) //||
 		)
 	{
 		fprintf(stderr,"ERROR WHILE ADDING TEST\n");

--- a/test/rofl/datapath/pipeline/tern_alike_test.c
+++ b/test/rofl/datapath/pipeline/tern_alike_test.c
@@ -3,7 +3,7 @@
 #include <assert.h>
 #include <arpa/inet.h>
 #include "CUnit/Basic.h"
-#include "../../../../../src/rofl/datapath/pipeline/common/ternary_fields.h"
+#include "rofl/datapath/pipeline/common/ternary_fields.h"
 
 int set_up(){
 	return 0;

